Frends.Community.Azure.Blob/UploadTask.cs

```diff
@@ -201,8 +206,16 @@ private static Encoding GetEncoding(string target)
             string fileName,
             CancellationToken cancellationToken)
         {
-            // get the destination blob, rename if necessary
-            var blob = new PageBlobClient(destinationProperties.ConnectionString, destinationProperties.ContainerName, fileName);
+            PageBlobClient blob;
+
+            if (destinationProperties.ConnectionMethod == ConnectionMethod.ConnectionString)
```

Comment:
This comment addresses all of blob and container client creation (in all tasks really). I see that all tasks have different classes that contain the same properties for connection. Can we extract an interface like
```
public interface IStorageAccountConnection
{
    enum Method = ConnectionString / ClientCredentials;
    string ConnectionString = ...
    string Clientid, secret, whatever else = ...
```
And then in some Util class make two methods like
```
class Utils
{
    public BlobClient CreateBlobClient(IStorageAccountConnection conn)
    { ... }

    public ContainerClient CreateContainerClient(IStorageAccountConnection conn)
    { ... }
}
```
This would remove quite a large portion of copy-pasted connection code. In every task you could just `Utils.CreateBlobClient(source)`.